From 2015f34fabee0c14617fa9347b9df6adc29d5569 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Date: Tue, 28 Feb 2017 16:45:46 -0500
Subject: [PATCH 2291/3082] drm/amd/display: Remove page_fleep_needed function.

This function makes a lot of assumptions to try and deduce
tha an atomic_commit came from FLIP ioctl. Those assumptions
are not explicit contract with DRM API and might lead to wrong
result as the code changes. Instead use drm_atomic_state.allow_modeset
flag which explicitly is set to false only in atomic flip
helper function, otherwise this flag is always true.
P.S When we get to do actual atomic IOCTL this will have to be revised
with the atomic call behavior.

Change-Id: I2c2c95daf015627a5f215fd8932a8a4fb4ba6d89
Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_types.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_types.c
index 31d6adb..e8aae53 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_types.c
@@ -2711,13 +2711,7 @@ void amdgpu_dm_atomic_commit_tail(
 			!crtc->state->active)
 			continue;
  
-                 pflip_needed = !state->allow_modeset &&
-                                 page_flip_needed(
-                                 plane_state,
-                                 old_plane_state,
-                                 crtc->state->event,
-                                 false);
- 
+	         pflip_needed = !state->allow_modeset; 
                  if (pflip_needed) {
                          amdgpu_dm_do_flip(
                                          crtc,
-- 
2.7.4

